libcloud/compute/drivers/linode.py

```diff
@@ -728,6 +752,912 @@ def _to_nodes(self, objs):
+
+    def _to_node(self, data):
+        extra = {
+            'tags': data['tags'],
+            'location': data['region'],
+            'ipv6': data['ipv6'],
+            'hypervisor': data['hypervisor'],
+            'specs': data['specs'],
+            'alerts': data['alerts'],
+            'backups': data['backups'],
+            'watchdog_enabled': data['watchdog_enabled']
+        }
+
+        public_ips = [ip for ip in data['ipv4'] if not is_private_subnet(ip)]
+        private_ips = [ip for ip in data['ipv4'] if is_private_subnet(ip)]
+        return Node(
+            id=data['id'],
+            name=data['label'],
+            state=self.LINODE_STATES[data['status']],
+            public_ips=public_ips,
+            private_ips=private_ips,
+            driver=self,
+            size=data['type'],
+            image=data['image'],
+            created_at=self._to_datetime(data['created']),
+            extra=extra)
+
+    def _to_datetime(self, strtime):
+        return datetime.strptime(strtime, "%Y-%m-%dT%H:%M:%S")
+
+    def _to_size(self, data):
+        extra = {
+            'class': data['class'],
+            'monthly_price': data['price']['monthly'],
+            'addons': data['addons'],
+            'successor': data['successor'],
+            'transfer': data['transfer'],
+            'vcpus': data['vcpus'],
+            'gpus': data['gpus']
+        }
+        return NodeSize(
+            id=data['id'],
+            name=data['label'],
+            ram=data['memory'],
+            disk=data['disk'],
+            bandwidth=data['network_out'],
+            price=data['price']['hourly'],
+            driver=self,
+            extra=extra
+        )
+
+    def _to_image(self, data):
+        extra = {
+            'type': data['type'],
+            'description': data['description'],
+            'created': self._to_datetime(data['created']),
+            'created_by': data['created_by'],
+            'is_public': data['is_public'],
+            'size': data['size'],
+            'eol': data['eol'],
+            'vendor': data['vendor'],
+        }
+        return NodeImage(
+            id=data['id'],
+            name=data['label'],
+            driver=self,
+            extra=extra
+        )
+
+    def _to_location(self, data):
+        extra = {
+            'status': data['status'],
+            'capabilities': data['capabilities'],
+            'resolvers': data['resolvers']
+        }
+        return NodeLocation(
+            id=data['id'],
+            name=data['id'],
+            country=data['country'].upper(),
+            driver=self,
+            extra=extra)
+
+    def _to_volume(self, data):
+        extra = {
+            'created': self._to_datetime(data['created']),
+            'tags': data['tags'],
+            'location': data['region'],
+            'linode_id': data['linode_id'],
+            'linode_label': data['linode_label'],
+            'state': self.LINODE_VOLUME_STATES[data['status']]
```

Comment:
filesystem_path' is an important value that is not returned in the extra fields:

'filesystem_path': data['filesystem_path']